### PR TITLE
refs #18919, fix validity of function scope

### PR DESCRIPTION
**Pull request by dylans**
Perhaps better than #249

### CODE REVIEW

File: mvc/_atBindingMixin.js [hunk 1]
```diff
@@ -7,15 +7,13 @@ define([
 	"./resolve",
 	"./sync"
 ], function(array, lang, declare, has, Stateful, resolve, sync){
-	if(has("mvc-bindings-log-api")){
-		function getLogContent(/*dojo/Stateful*/ target, /*String*/ targetProp){
+	var getLogContent = (has("mvc-bindings-log-api")) ? function getLogContent(/*dojo/Stateful*/ target, /*String*/ targetProp){
```
Comment (sindilevich) [inline mvc/_atBindingMixin.js]: @dylans, not that it has a difference, but did you make this function expression a named one on purpose?

**Comment by dylans** [comment]
@sindilevich thanks, was not intentional!

**Comment by sindilevich** [comment]
@dylans, @vansimke , I can't check it yet due to failing Java-based compilation after https://github.com/dojo/util/pull/55 changes. Need to resolve that first.

**Comment by vansimke** [comment]
@sindilevich what issues did dojo/util#55 cause? I didn't see discussion for a couple of days, so I merged that in. If there are other issues, please let me know and I'll address them as quickly as I can.

**Comment by sindilevich** [comment]
@vansimke, have reported the build errors in https://github.com/dojo/util/pull/55 itself.

**Comment by sindilevich** [comment]
@dylans, it seems that this PR do not cause any _compile-time_ problems. Checked with https://github.com/dojo/util/pull/56 changes.

File: mvc/_atBindingMixin.js [hunk 2]
```diff
@@ -7,15 +7,13 @@ define([
 	"./resolve",
 	"./sync"
 ], function(array, lang, declare, has, Stateful, resolve, sync){
-	if(has("mvc-bindings-log-api")){
-		function getLogContent(/*dojo/Stateful*/ target, /*String*/ targetProp){
+	var getLogContent = (has("mvc-bindings-log-api")) ? function (/*dojo/Stateful*/ target, /*String*/ targetProp){
```
Comment (sindilevich) [inline mvc/_atBindingMixin.js]: It looks you have an excessive white space here: `function (/*dojo/Stateful*/ target,`...

Comment (dylans) [inline mvc/_atBindingMixin.js]: yeah, spending too much time with Dojo 2 styling now... will fix now, thanks!

**Comment by sindilevich** [comment]
@dylans, looks OK to me. We're all waiting for Dojo 2 and Typescript. Getting Dojo 1.x with Typescript up and running is too large of a task.

**Comment by dylans** [comment]
Closed via 21c3bf8. Backported as b76a01c (1.11), 7a42a3d (1.10), and e27ef3b (1.9).

**Pull request closed without merging**